### ISSUE TITLE
on_result_diagnostics is running regardless of configuration for vscode tasks

### ISSUE THREAD

**Issue by willruggiano** [stevearc/overseer.nvim#163]
```lua
require("overseer").setup {
  component_aliases = {
    default = {
      { "display_duration", detail_level = 2 },
      "on_output_summarize",
      "on_exit_set_status",
      "on_complete_notify"
    },
    default_vscode = {
      "default",
      { "on_result_diagnostics_quickfix", open = true },
    },
  },
}
```

As you can see, `on_result_diagnostics` is not included in the default components for vscode tasks.

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "Build",
      "type": "typescript",
      "option": "build",
      "problemMatcher": "$tsc"
    }
  ]
}

```

And yet...

![image](https://github.com/stevearc/overseer.nvim/assets/11872440/86d519ba-0f25-4186-bf56-e68cd5a2e062)

**Comment by stevearc** [comment]
`on_result_diagnostics` is automatically added when you have a problemMatcher. This should generally be fine because all that component does is parse the output (as defined by the problemMatcher) and store that data on the task results. If you want, for example, to not add the diagnostics to the quickfix you can remove the `on_result_diagnostics_quickfix` component from the `default_vscode` alias.

What is the actual problem that you are trying to solve?

**Comment by willruggiano** [comment]
Adding them to the task results is fine, but I don't want them showing up, for example, with `vim.lsp.diagnostic.get_line_diagnostics()`. As you can see in the screenshot, this creates a redundant line diagnostic (the typescript lsp also reports the same diagnostic). Furthermore, if I fix the problem, the lsp diagnostic goes away but the overseer diagnostic does not. I know there is a `remove_on_restart` parameter for `on_result_diagnostics`, but that isn't exactly what I'm looking for.

TL;DR don't _show_ the diagnostic, I have LSPs for that. I just want the quickfix list so I can navigate the task result errors using `:cn`/`:cp`.

**Comment by stevearc** [comment]
Oh man, I definitely misspoke earlier: that component _does_ add diagnostics. You're absolutely right, I don't think we need to auto-add that component at all. Putting it in the `default_vscode` alias should be sufficient; I don't think anyone would ever need it to be _auto_-added.

**Comment by willruggiano** [comment]
Awesome, thanks!
